javaOOP_polymorphism/src/MainVehiclesExtension.java

```diff
@@ -0,0 +1,79 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+import jv.oop.exercise.vehicles.extension.Bus;
+import jv.oop.exercise.vehicles.extension.Car;
+import jv.oop.exercise.vehicles.extension.Truck;
+import jv.oop.exercise.vehicles.extension.Vehicle;
+
+/**
+ * Use your solution of the previous task for starting point and add more
+ * functionality. Add new vehicle – Bus. Now every vehicle has tank capacity and
+ * fuel quantity cannot fall (set) below 0 (If fuel quantity become less than 0
+ * print on the console "Fuel must be a positive number"). The vehicles cannot
+ * be filled with fuel more than their tank capacity. If you try to put more
+ * fuel in the tank than the available space, print on the console "Cannot fit
+ * fuel in tank" and do not add any fuel in vehicles tank. Add new command for
+ * the bus. The bus can drive with or without people. If the bus is driving with
+ * people, the air-conditioner is turned on and its fuel consumption per
+ * kilometer is increased with 1.4 liters. If there are no people in the bus
+ * when driving the air-conditioner is turned off and does not increase the fuel
+ * consumption. Input • On the first three lines you will receive information
+ * about the vehicles in format: Vehicle {initial fuel quantity} {liters per km}
+ * {tank capacity} • On fourth line - number of commands N that will be given on
+ * the next N lines • On the next N lines – commands in format o Drive Car
+ * {distance} o Drive Truck {distance} o Drive Bus {distance} o DriveEmpty Bus
+ * {distance} o Refuel Car {liters} o Refuel Truck {liters} o Refuel Bus
+ * {liters} Output • After each Drive command print whether the Car/Truck was
+ * able to travel given distance in format if it’s successful: "Car/Truck/Bus
+ * traveled {distance} km" • Or if it is not: "Car/Truck/Bus needs refueling" •
+ * If given fuel is ≤ 0 print "Fuel must be a positive number". • If given fuel
+ * cannot fit in car or bus tank print "Cannot fit fuel in tank"
+ * 
+ * • Finally print the remaining fuel for car, truck and bus rounded 2 digits
+ * after floating point in format: "Car: {liters} Truck: {liters} Bus: {liters}"
+ * 
+ * @author Ivelin Gavrailov
+ * @since 04.02.2021
+ *
+ */
+public class MainVehiclesExtension {
+
+	public static void main(String[] args) {
+		Scanner scan = new Scanner(System.in);
+		Map<String, Vehicle> vehicles = new HashMap<>();
+		String[] input = scan.nextLine().split("\\s+");
+		double fuelQuantity = Double.parseDouble(input[1]);
+		double fuelConsumption = Double.parseDouble(input[2]);
+		double tankCapacity = Double.parseDouble(input[2]);
+		vehicles.put("car", new Car(fuelQuantity, fuelConsumption, tankCapacity));
```

Comment:
I don't know if this is part of the task, but I see too much code repetition. You can create a method which reads a line, parse the values and creates one of the vehicles. Passing "car", "truck", "bus" as a paramether will make the method to know which vehicle instance to create.

Reply:
It was part of the task. but I decided to fix it anyway